src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpChangeNamespaceService.cs

```diff
@@ -19,13 +23,66 @@ namespace Microsoft.CodeAnalysis.CSharp.ChangeNamespace
     internal sealed class CSharpChangeNamespaceService :
         AbstractChangeNamespaceService<NamespaceDeclarationSyntax, CompilationUnitSyntax, MemberDeclarationSyntax>
     {
-        protected override SyntaxList<MemberDeclarationSyntax> GetMemberDeclarationsInContainer(SyntaxNode compilationUnitOrNamespaceDecl)
+        protected override async Task<ImmutableArray<(DocumentId, SyntaxNode)>> CanChangeNamespaceWorkerAsync(
+            Document document,
+            SyntaxNode container,
+            CancellationToken cancellationToken)
         {
-            if (compilationUnitOrNamespaceDecl is NamespaceDeclarationSyntax namespaceDecl)
+            if (document.Project.FilePath == null
+                || document.Project.Solution.Workspace.Kind == WorkspaceKind.MiscellaneousFiles
+                || document.IsGeneratedCode(cancellationToken))
+            {
+                return default;
+            }
+
+            TextSpan containerSpan;
+            if (container is NamespaceDeclarationSyntax)
+            {
+                containerSpan = container.Span;
+            }
+            else if (container is CompilationUnitSyntax)
+            {
+                // A compilation unit as container means user want to move all its members from global to some namespace.
+                // We use an empty span to indicate this case.
+                containerSpan = default;
+            }
+            else
+            {
+                return default;
```

Comment:
does this indicate a bug with teh caller?

Reply:
OK, will fix.

Are you done with the reviewing or you just started?

Reply:
I'm done reviewing.  it looks good tome :)